### DISCUSSION TITLE
Template config json file is generated on the first run creating template validation errors

### DISCUSSION THREAD

**Discussion by anshumanbh** [projectdiscovery/nuclei#3787] [Q&A]
### Nuclei version:
2.9.6

### Current Behavior:
As soon as I run Nuclei on a Linux box (`nuclei -h`), it creates a default config directory at $HOME/.config. Inside this directory, it creates a `.templates-config.json` file with the following contents:

```
{"nuclei-templates-directory":"/tmp/home/nuclei-templates","custom-s3-templates-directory":"/tmp/home/nuclei-templates/s3","custom-github-templates-directory":"/tmp/home/nuclei-templates/github","custom-gitlab-templates-directory":"/tmp/home/nuclei-templates/gitlab","custom-azure-templates-directory":"/tmp/home/nuclei-templates/azure","nuclei-latest-version":"","nuclei-templates-latest-version":""}
```
It specifies the default template directory to be `/tmp/home/nuclei-templates` as noted above. 
If I try to run the nuclei command such as `nuclei -u example.com -config <config-file> -validate`, it throws me a bunch of errors related to the below:

```
[ERR] Could not find template '/tmp/home/nuclei-templates': could not find file: stat /tmp/home/nuclei-templates: no such file or directory
```

My config file defines the path of the nuclei-templates directory that I want nuclei to use such as:
```
templates: /code/nuclei-templates/
workflows: /code/nuclei-templates/workflows/
```

But, this isn't respected by the `nuclei` command and for some reason it tries to resolve the templates against the `/tmp/home/` directory. 

### Expected Behavior:
We should at least be allowed to specify the path of our template-config.json file if we don't want to use the default one. Currently, there are no options in Nuclei that allows that. And, if the templates variable is specified in the config file and if the config file is passed to the `nuclei` command via the `config` flag, then it should really create the `.templates-config.json` file with that path, instead of the default one (`/tmp/home`)

### Steps To Reproduce:
1. Create a Nuclei config file and specify the template and workflow directory that you want to use
2. Run the nuclei command by simply typing `nuclei -h` and notice that a default config directory is created at $HOME/.config despite you providing it the config file

### Anything else:
This is very annoying and I have been struggling with it since a long time now. The default path of `/tmp/home/` is something that most likely is ephemeral every time an instance/container is booted and if we want to provide our own template directory, this behavior really makes it difficult to use Nuclei.

**Comment by ehsandeep** [comment]
@anshumanbh nuclei have options to overwrite all the default locations!

 First of all, you can use `nuclei -reset` option to remove previous set config, and you can use the below options to overwrite the default location where all the configs gets created, config files created by nuclei are meant to be used by nuclei and should be updated instead you can control the root directory location.
 
 To overwrite the default config directory location from `$HOME/.config/nuclei` to any directory, you can use `-config-directory` option.
 
 To overwrite nuclei-templates directory to install, you can use `-update-template-dir` to overwrite default location i.e `$HOME/nuclei-templates`
 
 
 ```
   -config-directory string       override the default config path ($home/.config)
   -ud, -update-template-dir string  custom directory to install / update nuclei-templates
```

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
